cmd/dogemap/main.go

```diff
@@ -1,21 +1,43 @@
 package main
 
 import (
-	"fmt"
 	"net/http"
-
-	"github.com/dogeorg/dogemap/pkg/dogemap"
+	"path/filepath"
+	"log"
+	"os"
+	"github.com/dogeorg/dogemap/pkg/dogemap/v2"
 )
 
 func main() {
-	// Start the HTTP server
-	go func() {
-		// Serve index.html using an HTTP server
-		http.Handle("/", http.FileServer(http.Dir("pkg/dogemap")))
-		fmt.Println("Server running at http://localhost:9090")
-		http.ListenAndServe(":9090", nil)
-	}()
+	staticFilesDir := "pkg/dogemap/v2"
+	indexFilePath := filepath.Join(staticFilesDir, "index.html")
+
+	// Create an http.FileSystem representing the root of the static files directory
+	fileSystem := http.Dir(staticFilesDir)
+
+	// File server to serve static files
+	fs := http.FileServer(fileSystem)
+
+	// Custom handler for routing to support Single Page Applications
+	// Specifically, its purpose is to serve the index.html whenever
+	// a 404 would be served.
+	http.HandleFunc("/", func(w http.ResponseWriter, r *http.Request) {
+		// Try to open the file directly using the file system
+		_, err := fileSystem.Open(r.URL.Path)
```

Comment:
It's a good idea to call Close() on the returned file handle if this call succeeds, to avoid leaking file handles

Reply:
lol, did not saw this, will fix it next :P